plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,185 @@
+package io.jenkins.plugins.casc;
+
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class SchemaGeneration {
+
+    final static JSONObject schemaTemplateObject = new JSONObject()
+        .put("$schema", "http://json-schema.org/draft-07/schema#")
+        .put("id", "http://jenkins.io/configuration-as-code#")
```

Comment:
Non-existent URL as ID? Why?

Reply:
It was picked up from the old schema :P IIRC we discussed somewhere to put the schema up on a site.@casz could  you please remind us?

Reply:
In the future we could properly look into using PCT and generating the full schema and put it on http://schemastore.org/json/